src/pages/activity/activity-planner/activity-duration/ActivityDuration.tsx

```diff
@@ -0,0 +1,69 @@
+import FlexCenter from "components/commons/FlexCenter/FlexCenter";
+import LoopSurveyPage from "components/commons/LoopSurveyPage/LoopSurveyPage";
+import { OrchestratorContext } from "interface/lunatic/Lunatic";
+import { callbackHolder, OrchestratorForStories } from "orchestrator/Orchestrator";
+import { useNavigate, useOutletContext, useParams } from "react-router-dom";
+import { EdtRoutesNameEnum } from "routes/EdtRoutesMapping";
+import { getLoopInitialPage, LoopEnum } from "service/loop-service";
+import { getNextLoopPage, getStepData } from "service/loop-stepper-service";
+import { getCurrentNavigatePath, getLoopParameterizedNavigatePath } from "service/navigation-service";
+import { saveData } from "service/survey-service";
+import { getActivities } from "service/survey-activity-service";
+
+const ActivityDurationPage = () => {
+    const navigate = useNavigate();
+    const context: OrchestratorContext = useOutletContext();
+    const currentPage = EdtRoutesNameEnum.ACTIVITY_DURATION;
+    const stepData = getStepData(currentPage);
+    const paramIteration = useParams().iteration;
+    const currentIteration = paramIteration ? +paramIteration : 0;
+    const activitiesAct = getActivities(context.idSurvey);
+
+    const specificProps = {
+        activitiesAct: activitiesAct,
+    };
+
+    const onNext = () => {
+        saveData(context.idSurvey, callbackHolder.getData()).then(() => {
+            navigate(
+                getLoopParameterizedNavigatePath(
+                    getNextLoopPage(currentPage),
+                    context.idSurvey,
+                    LoopEnum.ACTIVITY,
+                    currentIteration,
+                ),
+            );
+        });
+    };
+
+    const onClose = () => {
+        saveData(context.idSurvey, callbackHolder.getData()).then(() => {
+            navigate(getCurrentNavigatePath(context.idSurvey, EdtRoutesNameEnum.ACTIVITY, "3"));
```

Comment:
Perhaps the "3" might be put in a constant describing with it s name why it s a 3